framework/src/userobjects/ElementSubdomainModifier.C

```diff
@@ -34,6 +32,33 @@ ElementSubdomainModifier::validParams()
                                 "Boundary that associated with the unmoved domain when neighbor "
                                 "element(s) is moved. A boundary with the provided name will be "
                                 "created if not already exists on the mesh.");
+
+  params.addParam<std::vector<VariableName>>(
+      "initialize_variables",
+      {},
+      "Which variables to initialize when an element is moved into the active subdomains.");
```

Comment:
What happens if the user did not specify the IC for a specific variable in the activate subdomain(s)? Will the variable get trash initial value?